File: src/components/employeeService.js
import axios from 'axios';

class EmployeeService {

    async getAllEmployees() {
        try {
            const response = await axios.get('http://localhost:7142/api/employee');
            if (response.status === 200) {
                return response.data;
            } else {
                throw new Error(response.statusText);
            }
        } catch (error) {
            console.error(error);
            throw error;
        }
    }

    async searchEmployees(searchTerm) {
        try {
            const response = await axios.get(`http://localhost:7142/api/employee/search?searchTerm=${searchTerm}`);
            if (response.status === 200) {
                return response.data;
            } else {
                throw new Error(response.statusText);
            }
        } catch (error) {
            console.error(error);
            throw error;
        }
    }
        async insertEmployee(employee) {
            try {
                const response = await axios.post('http://localhost:7142/api/employee', employee);
                if (response.status === 201) {
                    return response.data;
                } else {
                    throw new Error(response.statusText);
                }
            } catch (error) {
                console.error(error);
                throw error;
            }
        }
    
        async deleteEmployee(id) {
            try {
                const response = await axios.delete(`http://localhost:7142/api/employee/${id}`);
                if (response.status === 200) {
                    return response.data;
                } else {
                    throw new Error(response.statusText);
                }
            } catch (error) {
                console.error(error);
                throw error;
            }
        }
    
        async updateEmployee(employee) {
            try {
                const response = await axios.put(`http://localhost:7142/api/employee/${employee.id}`, employee);
                if (response.status === 200) {
                    return response.data;
                } else {
                    throw new Error(response.statusText);
                }
            } catch (error) {
                console.error(error);
                throw error;
            }
        }
    
}
export default EmployeeService;

File: Phone Directory/Services/EmployeeService.cs
using Phone_Directory.Models;
using PhoneDirectory.Data;


namespace PhoneDirectory.Data
{
    public class EmployeeService : IEmployeeService
    {
        private readonly EmployeeContext _context;
        private readonly IEmployeeRepository _employeeRepository;

        public EmployeeService(EmployeeContext context, IEmployeeRepository employeeRepository)
        {
            _context = context;
            _employeeRepository = employeeRepository;
        }

        public List<Employee> GetAllEmployees()
        {
            return _employeeRepository.GetAllEmployees();
        }

        public List<Employee> SearchEmployees(string searchTerm)
        {
            return _employeeRepository.SearchEmployees(searchTerm);
        }

        public void InsertEmployee(AddEmployee employee)
        {
            _employeeRepository.InsertEmployee(employee);
            _context.SaveChanges();
        }

        public void UpdateEmployee(Employee employee)
        {
            _employeeRepository.UpdateEmployee(employee);
            _context.SaveChanges();
        }

        public void DeleteEmployee(int id)
        {
            _employeeRepository.DeleteEmployee(id);
            _context.SaveChanges();
        }


    }
}
